tests/contrib/hooks/test_livy_hook.py

```diff
@@ -0,0 +1,428 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+from unittest.mock import patch, MagicMock
+import json
+from requests.exceptions import RequestException
+
+from airflow import AirflowException
+from airflow.models import Connection
+from airflow.utils import db
+
+from airflow.contrib.hooks.livy_hook import LivyHook, BatchState
+
+TEST_ID = 100
+SAMPLE_GET_RESPONSE = {'id': TEST_ID, 'state': BatchState.SUCCESS.value}
+
+
+class TestLivyHook(unittest.TestCase):
+
+    def setUp(self):
```

Comment:
I think [setUpClass](https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUpClass) would also work and it wouldn't call those "db merges" before each test. And you can specify [tearDownClass](https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDownClass) to drop the connections.